src/core/layout/qgslayoutitemmap.cpp

```diff
@@ -1155,6 +1155,14 @@ QgsExpressionContext QgsLayoutItemMap::createExpressionContext() const
   scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "map_crs_definition" ), mapCrs.toProj4(), true ) );
   scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "map_units" ), QgsUnitTypes::toString( mapCrs.mapUnits() ), true ) );
 
+  QVariantList layers;
+  QList<QgsMapLayer *> layersInMap = layersToRender( &context );
+  for ( QgsMapLayer *layer : layersInMap )
+  {
+    layers << layer->id();
```

Comment:
Can you add the layer directly here?

See qgsexpressioncontext.cpp
```
scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "layer" ), QVariant::fromValue<QgsWeakMapLayerPointer >( QgsWeakMapLayerPointer( const_cast<QgsMapLayer *>( layer ) ) ), true, true ) );
```

If required you can also add another one with `map_layer_ids`

Reply:
Yes, I will add `map_layers` for  the layer directly.
And `map_layers_ids` for ids. I need it for `layer_property(layer,property)` which takes a layer name or layer id.

Reply:
I think `layer_property` should also work directly with a layer object, no?

Reply:
Not according to the document ion ;-)
https://github.com/qgis/QGIS/blob/master/resources/function_help/json/layer_property#L6

I will try

Reply:
The docs are outdated here, a direct layer object is permissible